src/Shared/ServerInfrastructure/BufferExtensions.cs

```diff
@@ -134,29 +134,28 @@ internal static void WriteNumeric(ref this BufferWriter<PipeWriter> buffer, ulon
         const byte AsciiDigitStart = (byte)'0';
 
         var start = buffer.Span;
-        var bytesLeftInBlock = start.Length;
 
         // Fast path, try copying to the available memory directly
         var simpleWrite = true;
-        if (number < 10 && bytesLeftInBlock >= 1)
+        if (number < 10 && start.Length >= 1)
         {
             start[0] = (byte)(((uint)number) + AsciiDigitStart);
             buffer.Advance(1);
         }
-        else if (number < 100 && bytesLeftInBlock >= 2)
+        else if (number < 100 && start.Length >= 2)
         {
             var val = (uint)number;
-            var tens = (byte)((val * 205u) >> 11); // div10, valid to 1028
+            uint tens = (byte)((val * 205u) >> 11); // div10, valid to 1028
```

Comment:
Here I don't know...

ASP.NET Core has the convention to use `var`, so 
```suggestion
            var tens = (uint)(byte)((val * 205u) >> 11); // div10, valid to 1028
```

Personally I prefer the explicit type, so as you've written it.

Reply:
Updated, although there was no compiler warning (the silent `...`) here, hence I went with as you suggested earlier.